src/pages/user/InstitutionSelector.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  emptyInstitution,
+  Institution,
+  InstitutionSubUnit,
+  emptyInstitutionSubUnit,
+} from '../../types/institution.types';
+import InstitutionSearch from '../publication/references_tab/components/InstitutionSearch';
+import SubUnitSelect from './SubUnitSelect';
+import { getInstitutionSubUnit } from '../../api/institutionApi';
+import styled from 'styled-components';
+import InstitutionPresentation from './InstitutionPresentation';
+
+const StyledInstitutionSelector = styled.div`
+  width: 30rem;
+`;
+
+interface InstitutionSelectorProps {
+  setSelectedCrustinUnitId: (value: string) => void;
+}
+
+const InstitutionSelector: React.FC<InstitutionSelectorProps> = ({ setSelectedCrustinUnitId }) => {
+  const { t } = useTranslation('profile');
+  const [faculties, setFaculties] = useState<InstitutionSubUnit[]>([]);
+  const [institutes, setInstitutes] = useState<InstitutionSubUnit[]>([]);
+  const [selectedInstitution, setSelectedInstituion] = useState(emptyInstitution);
+  const [selectedFaculty, setSelectedFaculty] = useState(emptyInstitutionSubUnit);
+  const [selectedInstitute, setSelectedInstitute] = useState(emptyInstitutionSubUnit);
+
+  const getSubUnits = async (searchValue: string, setValueFunction: (value: any) => void) => {
```

Comment:
kalle setValueFunction for `setStateFunction` ? og legge på typer på value? f.eks 
`value: Partial<Institution> | Partial<InstitutionSubUnit>`